src/sage/graphs/base/boost_interface.cpp

```diff
@@ -84,11 +84,11 @@ class BoostGraph
     // This map is a parameter/output for biconnected_components function
     typedef typename std::map<edge_descriptor, int, order_edges> edge_map;
 
-public:
     adjacency_list graph;
     std::vector<vertex_descriptor> vertices;
     vertex_to_int_map index;
 
+public:
```

Comment:
This is not necessary to fix the warning. But since these fields are not part of the Cython interface, they can as well be private (would have helped me to understand things more quickly.